### ISSUE TITLE
Scss string interpolation as custom property default

### ISSUE THREAD

**Issue by pascalduez** [stylelint/stylelint#4999]
> Clearly describe the bug

Using an SCSS string interpolation for a custom property `var()` function default produce a `CSSSyntaxError`.

> What code is needed to reproduce the bug?

```css
$color: #fff;

.test {
  color: var(--color, #{$color});
}
```

> Which version of stylelint are you using?

`13.7.2`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint --syntax scss 'src/**/*.scss'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS string interpolation.

> What did you expect to happen?

No error.

> What actually happened (e.g. what warnings or errors did you get)?

The following error were flagged:

```shell
$ stylelint --syntax scss 'src/**/*.scss'

<input css 1>
 1:11  ✖  Unknown word   CssSyntaxError
```

**Comment by pascalduez** [comment]
Narrowed down to `stylelint-value-no-unknown-custom-properties` plugin.

config:
```json
{
  "plugins": ["stylelint-value-no-unknown-custom-properties"],
  "rules": {
    "csstools/value-no-unknown-custom-properties": [
      true,
      {
        "importFrom": "src/styles/global/variables.css"
      }
    ]
  }
}
```
